lib/katello/engine.rb

```diff
@@ -4,11 +4,13 @@ class Engine < ::Rails::Engine
 
     initializer 'katello.selective_params_parser', :before => :build_middleware_stack do |app|
       require 'katello/params_parser_wrapper'
-      app.middleware.swap(
-        ActionDispatch::ParamsParser,
-        Katello::ParamsParserWrapper,
-        ->(env) { env['PATH_INFO'] =~ /consumers/ && env['PATH_INFO'] =~ /profile|packages/ }
-      )
+      if Rails::VERSION::MAJOR == 4
+        app.middleware.swap(
+          ActionDispatch::ParamsParser,
+          Katello::ParamsParserWrapper,
+          ->(env) { env['PATH_INFO'] =~ /consumers/ && env['PATH_INFO'] =~ /profile|packages/ }
+        )
+      end
```

Comment:
I see, I think we would have to extend ActionDispatch::Request::Utils::ParamEncoder now to do it https://github.com/rails/rails/blob/310918f6a194bf5752fe1025930881756f5d8a8e/actionpack/lib/action_dispatch/request/utils.rb